tough/src/editor/mod.rs

```diff
@@ -44,28 +51,40 @@ const SPEC_VERSION: &str = "1.0.0";
 /// `sign()` method, which takes a given set of signing keys, builds each of
 /// the roles using the data provided, and signs the roles. This results in a
 /// `SignedRepository` which can be used to write the repo to disk.
+///
+/// `RepositoryEditor` uses a modal design to edit `Targets`. `TargetsEditor`
+/// is used to perform all actions on a specified `Targets`. To change the
+/// `Targets` being used call `change_delegated_targets()` to create a new `TargetsEditor`
+/// for the specified role. To sign a `Targets` role from the `TargetsEditor` use `sign_targets_editor()`.
+/// This will clear out the targets editor and insert the newly signed targets in `signed_targets`.
+///
+/// To update an existing targets from a metadata file use `update_delegated_targets()`.
+///
+/// To add a new role from metadata to the `Targets` in `TargetsEditor` use `add_role()`.
 #[derive(Debug)]
-pub struct RepositoryEditor {
+pub struct RepositoryEditor<'a, T: Transport> {
```

Comment:
Can we own the transport instead of using a reference so we don't need the lifetime?  Can we Box it instead of the generic so we don't have to add the generic here either?  I agree that it's confusing...

Reply:
I'm not sure what we can do. I know that Repository is set up this way.